state/state.go

```diff
@@ -793,6 +809,17 @@ func (s *State) fetchGuild(id discord.GuildID) (g *discord.Guild, err error) {
 	return
 }
 
+func (s *State) fetchRoles(gID discord.GuildID) (rs []discord.Role, err error) {
+	rs, err = s.Session.Roles(gID)
+	if err == nil && s.HasIntents(gateway.IntentGuilds) {
```

Comment:
With this check, we're better off just returning in `Roles()` if the intents are not there. Then, this check simply becomes `if err == nil {`.

The same should also apply for `fetch{Guild,Member}`.

Reply:
Nevermind, I misread the code. Carry on.